c7n/resources/redshift.py

```diff
@@ -400,6 +400,49 @@ def process_vpc_routing(self, cluster):
                 EnhancedVpcRouting=new_routing)
 
 
+@actions.register('set-public-access')
+class RedshiftSetPublicAccess(BaseAction):
+    """
+    Action to set the 'PubliclyAccessible' setting on a redshift cluster
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+                - name: redshift-set-public-access
+                  resource: redshift
+                  filters:
+                    - PubliclyAccessible: true
+                  actions:
+                    - type: set-public-access
+                      state: false
+    """
+
+    schema = type_schema(
+        'set-public-access',
+        state={'type': 'boolean'})
+    permissions = ('redshift:ModifyCluster',)
+
+    def set_access(self, c):
+        client = local_session(self.manager.session_factory).client('redshift')
+        client.modify_cluster(
+            ClusterIdentifier=c['ClusterIdentifier'],
+            PubliclyAccessible=self.data.get('state', False))
+
+    def process(self, clusters):
+        with self.executor_factory(max_workers=2) as w:
+            futures = []
+            for c in clusters:
+                futures.append(w.submit(self.set_access, c))
+            for f in as_completed(futures):
+                if f.exception():
+                    self.log.error(
+                        "Exception changing Redshift public access  \n %s",
```

Comment:
can you log the cluster id here, basically switch futures to a dictionary to associate the future to the cluster, and then derereference and log. else the log isn't really helpful in identifying an issue.